libraries/AP_HAL/hwdef/scripts/hwdef.py

```diff
@@ -307,6 +311,14 @@ def process_line_define(self, line, depth, a):
             name = result.group(1)
             self.assert_good_define(name)
 
+        # extract all defines as name/value pairs
+        result = re.match(r'define\s*([A-Z_0-9]+)\s*(.*)', line)
+        if result is not None:
+            value = result.group(2).strip()
+            if value is not None and len(value) == 0:
+                value = None
+            self.defines[result.group(1)] = value
+
```

Comment:
Merge with the above (you can use an optional capture)

Reply:
Done.

I removed the intdefines match while I was at it.